File: apps/android/core/src/main/kotlin/diy/atelier/core/AppProduct.kt
package diy.atelier.core

enum class AppProduct(
    val displayName: String,
    val applicationId: String,
    val sections: List<WorkspaceSection>,
) {
    ATELIER(
        displayName = "Atelier",
        applicationId = "diy.atelier",
        sections = listOf(
            WorkspaceSection("today", "Today", "T"),
            WorkspaceSection("inbox", "Inbox", "I"),
            WorkspaceSection("projects", "Projects", "P"),
        ),
    ),
    NOTES(
        displayName = "Atelier Notes",
        applicationId = "diy.atelier.notes",
        sections = listOf(
            WorkspaceSection("all-notes", "All Notes", "N"),
            WorkspaceSection("recent", "Recent", "R"),
            WorkspaceSection("shared", "Shared", "S"),
        ),
    ),
    MAIL(
        displayName = "Atelier Mail",
        applicationId = "diy.atelier.mail",
        sections = listOf(
            WorkspaceSection("inbox", "Inbox", "I"),
            WorkspaceSection("starred", "Starred", "★"),
            WorkspaceSection("sent", "Sent", "S"),
        ),
    ),
    CALENDAR(
        displayName = "Atelier Calendar",
        applicationId = "diy.atelier.calendar",
        sections = listOf(
            WorkspaceSection("upcoming", "Upcoming", "U"),
            WorkspaceSection("invitations", "Invitations", "I"),
            WorkspaceSection("feeds", "Calendar Feeds", "F"),
        ),
    ),
    TASKS(
        displayName = "Atelier Tasks",
        applicationId = "diy.atelier.tasks",
        sections = listOf(
            WorkspaceSection("inbox", "Inbox", "I"),
            WorkspaceSection("today", "Today", "T"),
            WorkspaceSection("projects", "Projects", "P"),
        ),
    ),
}

data class WorkspaceSection(
    val id: String,
    val title: String,
    val shortLabel: String,
)

File: apps/apple/Packages/AtelierShared/Sources/AtelierCore/AppProduct.swift
import Foundation

public enum AppProduct: String, CaseIterable, Codable, Identifiable, Sendable {
    case atelier
    case notes
    case mail
    case calendar
    case tasks

    public var id: String { rawValue }

    public var displayName: String {
        switch self {
        case .atelier: "Atelier"
        case .notes: "Atelier Notes"
        case .mail: "Atelier Mail"
        case .calendar: "Atelier Calendar"
        case .tasks: "Atelier Tasks"
        }
    }

    public var bundleIdentifier: String {
        switch self {
        case .atelier: "diy.atelier"
        case .notes: "diy.atelier.notes"
        case .mail: "diy.atelier.mail"
        case .calendar: "diy.atelier.calendar"
        case .tasks: "diy.atelier.tasks"
        }
    }

    public var systemImage: String {
        switch self {
        case .atelier: "square.grid.2x2"
        case .notes: "note.text"
        case .mail: "envelope"
        case .calendar: "calendar"
        case .tasks: "checkmark.circle"
        }
    }

    public var sections: [WorkspaceSection] {
        switch self {
        case .atelier:
            [
                .init(id: "today", title: "Today", systemImage: "sun.max"),
                .init(id: "inbox", title: "Inbox", systemImage: "tray"),
                .init(id: "projects", title: "Projects", systemImage: "folder"),
            ]
        case .notes:
            [
                .init(id: "all-notes", title: "All Notes", systemImage: "note.text"),
                .init(id: "recent", title: "Recent", systemImage: "clock"),
                .init(id: "shared", title: "Shared", systemImage: "person.2"),
            ]
        case .mail:
            [
                .init(id: "inbox", title: "Inbox", systemImage: "tray"),
                .init(id: "starred", title: "Starred", systemImage: "star"),
                .init(id: "sent", title: "Sent", systemImage: "paperplane"),
            ]
        case .calendar:
            [
                .init(id: "upcoming", title: "Upcoming", systemImage: "calendar"),
                .init(id: "invitations", title: "Invitations", systemImage: "envelope.open"),
                .init(id: "feeds", title: "Calendar Feeds", systemImage: "dot.radiowaves.left.and.right"),
            ]
        case .tasks:
            [
                .init(id: "inbox", title: "Inbox", systemImage: "tray"),
                .init(id: "today", title: "Today", systemImage: "sun.max"),
                .init(id: "projects", title: "Projects", systemImage: "list.bullet.rectangle"),
            ]
        }
    }
}

public struct WorkspaceSection: Identifiable, Hashable, Codable, Sendable {
    public let id: String
    public let title: String
    public let systemImage: String

    public init(id: String, title: String, systemImage: String) {
        self.id = id
        self.title = title
        self.systemImage = systemImage
    }
}
